src/lib/storage/yaml_writer.rb

```diff
@@ -139,7 +158,7 @@ def yaml_partition(partition)
           "size" => DiskSize.new(partition.region.to_kb(partition.region.length)).to_s,
           "name" => partition.name,
           "type" => @partition_types[partition.type],
-          "id"   => @partition_ids[partition.id]
+          "id"   => @partition_ids[partition.id] || "0x#{partition.id.to_s(16)}"
```

Comment:
NP: this code is quite confusing, why partition.id is not id and need additional mapping? and what means default here? hexa code of partition_id